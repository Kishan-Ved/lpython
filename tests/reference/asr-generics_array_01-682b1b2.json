{
    "basename": "asr-generics_array_01-682b1b2",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/generics_array_01.py",
    "infile_hash": "6e943dd0e26ab4d1ffb6ed6909a365b4135b6f5295957b2478cfb479",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-generics_array_01-682b1b2.stdout",
    "stdout_hash": "49a0ae466eca111a4439e97c3c285259e077b40f20019b3fe15fd7e0",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}