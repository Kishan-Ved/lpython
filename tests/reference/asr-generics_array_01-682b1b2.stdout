(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            T:
                                (Variable
                                    2
                                    T
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (TypeParameter
                                        T
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            __asr_generic_f_0:
                                (Function
                                    (SymbolTable
                                        217
                                        {
                                            _lpython_return_variable:
                                                (Variable
                                                    217
                                                    _lpython_return_variable
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            i:
                                                (Variable
                                                    217
                                                    i
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            lst:
                                                (Variable
                                                    217
                                                    lst
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Integer 4)
                                                        [(()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    __asr_generic_f_0
                                    (FunctionType
                                        [(Array
                                            (Integer 4)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                        (Integer 4)]
                                        (Integer 4)
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 217 lst)
                                    (Var 217 i)]
                                    [(=
                                        (ArrayItem
                                            (Var 217 lst)
                                            [(()
                                            (IntegerConstant 0 (Integer 4))
                                            ())]
                                            (Integer 4)
                                            RowMajor
                                            ()
                                        )
                                        (Var 217 i)
                                        ()
                                    )
                                    (=
                                        (Var 217 _lpython_return_variable)
                                        (ArrayItem
                                            (Var 217 lst)
                                            [(()
                                            (IntegerConstant 0 (Integer 4))
                                            ())]
                                            (Integer 4)
                                            RowMajor
                                            ()
                                        )
                                        ()
                                    )
                                    (Return)]
                                    (Var 217 _lpython_return_variable)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            __main__global_stmts:
                                (Function
                                    (SymbolTable
                                        218
                                        {
                                            
                                        })
                                    __main__global_stmts
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [use_array]
                                    []
                                    [(SubroutineCall
                                        2 use_array
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            f:
                                (Function
                                    (SymbolTable
                                        215
                                        {
                                            _lpython_return_variable:
                                                (Variable
                                                    215
                                                    _lpython_return_variable
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (TypeParameter
                                                        T
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            i:
                                                (Variable
                                                    215
                                                    i
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (TypeParameter
                                                        T
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            lst:
                                                (Variable
                                                    215
                                                    lst
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (TypeParameter
                                                            T
                                                        )
                                                        [(()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    f
                                    (FunctionType
                                        [(Array
                                            (TypeParameter
                                                T
                                            )
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                        (TypeParameter
                                            T
                                        )]
                                        (TypeParameter
                                            T
                                        )
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 215 lst)
                                    (Var 215 i)]
                                    [(=
                                        (ArrayItem
                                            (Var 215 lst)
                                            [(()
                                            (IntegerConstant 0 (Integer 4))
                                            ())]
                                            (TypeParameter
                                                T
                                            )
                                            RowMajor
                                            ()
                                        )
                                        (Var 215 i)
                                        ()
                                    )
                                    (=
                                        (Var 215 _lpython_return_variable)
                                        (ArrayItem
                                            (Var 215 lst)
                                            [(()
                                            (IntegerConstant 0 (Integer 4))
                                            ())]
                                            (TypeParameter
                                                T
                                            )
                                            RowMajor
                                            ()
                                        )
                                        ()
                                    )
                                    (Return)]
                                    (Var 215 _lpython_return_variable)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            use_array:
                                (Function
                                    (SymbolTable
                                        216
                                        {
                                            array:
                                                (Variable
                                                    216
                                                    array
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Integer 4)
                                                        [((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 1 (Integer 4)))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            x:
                                                (Variable
                                                    216
                                                    x
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    use_array
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [__asr_generic_f_0]
                                    []
                                    [(=
                                        (Var 216 array)
                                        (ArrayConstant
                                            []
                                            (Array
                                                (Integer 4)
                                                [((IntegerConstant 0 (Integer 4))
                                                (IntegerConstant 1 (Integer 4)))]
                                                FixedSizeArray
                                            )
                                            RowMajor
                                        )
                                        ()
                                    )
                                    (=
                                        (Var 216 x)
                                        (IntegerConstant 69 (Integer 4))
                                        ()
                                    )
                                    (Print
                                        [(FunctionCall
                                            2 __asr_generic_f_0
                                            ()
                                            [((ArrayPhysicalCast
                                                (Var 216 array)
                                                FixedSizeArray
                                                DescriptorArray
                                                (Array
                                                    (Integer 4)
                                                    [((IntegerConstant 0 (Integer 4))
                                                    (IntegerConstant 1 (Integer 4)))]
                                                    DescriptorArray
                                                )
                                                ()
                                            ))
                                            ((Var 216 x))]
                                            (Integer 4)
                                            ()
                                            ()
                                        )]
                                        ()
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    __main__
                    [numpy]
                    .false.
                    .false.
                ),
            lpython_builtin:
                (IntrinsicModule lpython_builtin),
            main_program:
                (Program
                    (SymbolTable
                        219
                        {
                            __main__global_stmts:
                                (ExternalSymbol
                                    219
                                    __main__global_stmts
                                    2 __main__global_stmts
                                    __main__
                                    []
                                    __main__global_stmts
                                    Public
                                )
                        })
                    main_program
                    [__main__]
                    [(SubroutineCall
                        219 __main__global_stmts
                        2 __main__global_stmts
                        []
                        ()
                    )]
                ),
            numpy:
                (Module numpy)
        })
    []
)
